cmd/tailscale/cli/file.go

```diff
@@ -286,22 +287,106 @@ func runCpTargets(ctx context.Context, args []string) error {
 	return nil
 }
 
+type onConflict string
+
+const (
+	skipOnExist         onConflict = "skip"
+	overwriteExisting              = "overwrite" //  Overwrite any existing file at the target location
+	createNumberedFiles            = "rename"    //  Create an alternately named file in the style of Chrome Downloads
+)
+
+func (v *onConflict) String() string {
+	return string(*v)
+}
+
+func (v *onConflict) Set(s string) error {
+	if s == "" {
+		*v = skipOnExist
+		return nil
+	}
+	l := onConflict(strings.ToLower(s))
+	if l != skipOnExist && l != overwriteExisting && l != createNumberedFiles {
+		return fmt.Errorf("'%s' not one of (skip|overwrite|rename)", s)
+	}
+	*v = l
+	return nil
+}
+
 var fileGetCmd = &ffcli.Command{
 	Name:       "get",
-	ShortUsage: "file get [--wait] [--verbose] <target-directory>",
+	ShortUsage: "file get [--wait] [--verbose] [--conflict=(skip|overwrite|rename)] <target-directory>",
 	ShortHelp:  "Move files out of the Tailscale file inbox",
 	Exec:       runFileGet,
 	FlagSet: (func() *flag.FlagSet {
 		fs := newFlagSet("get")
 		fs.BoolVar(&getArgs.wait, "wait", false, "wait for a file to arrive if inbox is empty")
 		fs.BoolVar(&getArgs.verbose, "verbose", false, "verbose output")
+		fs.Var(&getArgs.conflict, "conflict", "behavior when target file already exists")
 		return fs
 	})(),
 }
 
-var getArgs struct {
-	wait    bool
-	verbose bool
+var getArgs = struct {
+	wait     bool
+	verbose  bool
+	conflict onConflict
+}{conflict: skipOnExist}
+
+func numberedFileName(dir string, name string, i int) string {
+	extension := path.Ext(name)
+	numberedFileName := fmt.Sprintf("%s (%d)%s", name[0:len(name)-len(extension)], i, extension)
+	return filepath.Join(dir, numberedFileName)
+}
+
+func receiveFile(ctx context.Context, wf apitype.WaitingFile, dir string) (targetFile string, size int64, err error) {
+	rc, size, err := tailscale.GetWaitingFile(ctx, wf.Name)
+	targetFile = filepath.Join(dir, wf.Name)
+	if err != nil {
+		err = fmt.Errorf("opening inbox file %q: %w", wf.Name, err)
+		return
+	}
+	of, err := os.OpenFile(targetFile, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0644)
```

Comment:
I think the rest of this would be simpler if you pulled this part out into its own func, passing in targetFile and getArgs.conflict and returning what OpenFile already returns

Reply:
done